docs/specs/fee-calculator.md

```diff
@@ -0,0 +1,405 @@
+3. **Filter Distributors**: If distributorAddress provided, filter to single distributor
+4. **For Each Distributor**:
+   - Load balance history from FileManager
+   - Load distribution events from FileManager
+   - Build chronological timeline of all dates with data
+   - Sort balance entries by date
+   - For each date with balance data:
+     - If first date: balance change = 0 (no previous balance)
+     - Otherwise: balance change = current balance - previous balance
+     - Find all distribution events for this date
+     - Sum distribution amounts: total distributions = sum of all values
+     - Calculate daily fee = balance change + total distributions
+     - Store result with date, balances, distributions, and fee
+5. **Generate Report**: Compile all distributor results into structured report
+6. **Write Results**: Save report using FileManager
+
+### Data Validation
+
+- Ensure all dates are in ISO format (YYYY-MM-DD)
+- Verify balance values are valid numeric strings
+- Confirm distribution amounts are non-negative
+- Check for data gaps and report them without failing
+
+### Edge Case Handling
+
+1. **First Day**: No previous balance, so balance change = 0
+2. **Missing Events**: If no events for a date, distributions = 0
+3. **Data Gaps**: Report gaps but continue processing available data
+4. **Zero Balances**: Valid scenario, process normally
+
+## 7. Data Structures
+
+### Input Types
+
+```typescript
+interface DistributorsData {
+  distributors: Array<{
+    address: string;
+    metadata: {
+      name: string;
+      created_at: string;
+    };
+  }>;
+}
+
+interface BalanceData {
+  metadata: {
+    chain_id: number;
+    reward_distributor: string;
+  };
+  balances: {
+    [date: string]: {
+      block_number: number;
+      balance_wei: string;
+    };
+  };
+}
+
+interface RecipientRecievedEventData {
+  metadata: {
+    chain_id: number;
+    distributor_address: string;
+  };
+  events: Array<{
+    block_number: number;
+    transaction_hash: string;
+    log_index: number;
+    timestamp: string;
+    date: string;
+    recipient: string;
+    amount_wei: string;
+  }>;
+}
+```
+
+### Output Types
+
+```typescript
+interface FeeReport {
+  metadata: {
+    chain_id: number;
+    generated_at: string;
+    report_type: "daily_fees";
+  };
+  distributors: {
+    [distributorAddress: string]: {
+      name: string;
+      daily_fees: Array<{
+        date: string;
+        start_balance_wei: string;
+        end_balance_wei: string;
+        balance_change_wei: string;
+        distributions_wei: string;
+        distributions_count: number;
+        fee_wei: string;
+      }>;
+    };
+  };
+}
+```
+
+### Internal Calculation Types
+
+```typescript
+interface DailyFeeCalculation {
+  date: string;
+  startBalance: bigint;
+  endBalance: bigint;
+  balanceChange: bigint;
+  distributions: bigint;
+  distributionCount: number;
+  totalFee: bigint;
+}
+```
+
+## 8. Implementation Requirements
+
+### Validation Rules
```

Comment:
Get rid of this section.

Reply:
you still need to remove this whole section about data validation - we dont want to do this